linked_list/linked_list.py

```diff
@@ -10,70 +10,145 @@ def __init__(self, value, next_node = None):
+        counter = 0
+        current = self.head
+        while current:
+            counter += 1
+            current = current.next
+        return counter
+
 
     # method that returns the value at a given index in the linked list
     # index count starts at 0
     # returns None if there are fewer nodes in the linked list than the index value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def get_at_index(self, index):
-        pass
+        current = self.head
+        if current == None:
+            return None
+        counter = 0
+        while counter != index:
+            current = current.next
+            counter += 1
+        return current.value 
+            
 
     # method that returns the value of the last node in the linked list
     # returns None if the linked list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(n)
+    
     def get_last(self):
-        pass
+        current = self.head
+        if not current:
+            return None
+        while current.next:
+            current = current.next
+        return current.value
 
     # method that inserts a given value as a new last node in the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: 0(n) because it has to move through each existing node
+    # Space Complexity: O(n)
+    
     def add_last(self, value):
-        pass
+        new_node = Node(value)
+        if self.head == None:
+            self.head = new_node
+            return
+        current_node = self.head
+        while current_node.next:
+           current_node = current_node.next
+        current_node.next = new_node
+            
 
     # method to return the max value in the linked list
     # returns the data value and not the node
     def find_max(self):
-        pass
+        if not self.head:
+            return None
+        current_node = self.head
+        max = self.head.value
+        while current_node:
+            if current_node.value > max:
+                max = current_node.value
+                current_node = current_node.next
+            else:
+                current_node = current_node.next
+        return max
 
     # method to delete the first node found with specified value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def delete(self, value):
-        pass
+        current_node = self.head
+        if not current_node:
+            return None
+        if current_node.next.value == value:
+            current_node = self.head
+        
+        while current_node.next:
+            if current_node.next.value == value:
+                if not current_node.next.next:
+                    current_node.next = None
+                else:
+                    current_node = current_node.next.next
+                    return

```

Comment:
The inner `if` is not needed

```suggestion
            if current_node.next.value == value:
                current_node = current_node.next.next
                return
```